internal/engine/wazevo/testcases/testcases.go

```diff
@@ -2318,6 +2318,50 @@ var (
 			}}},
 		},
 	}
+
+	AtomicCasConst0 = TestCase{
```

Comment:
I added this test case to check the behavior regards to

https://github.com/tetratelabs/wazero/blob/main/internal/engine/compiler/impl_threads_arm64.go#L411

I am assuming `tmp := operandNR(m.compiler.AllocateVReg(si.Return().Type()))` and the `mov` is what ensures we don't run into zero register issues with this backend but let me know if that's not the case